### ISSUE TITLE
cache resources exhausted Error

### ISSUE THREAD

**Issue by asitemade4u** [ropensci/magick#366]
- I am using the magick package, among other things, to watermark images in batch.
- The R environment from which magick is called is a rocker docker container, itself called by Drone CI.

---
Drone CI -> rocker container -> magick
---

- I have mapped, within the rocker container, the magick `tmp` folder to a folder on the host where there is planty of space.
- I also have changed the `/etc/ImageMagick6/policy.xml` file within the rocker container to accommodate more memory and set the Magick `tmp` folder to the folder mapped above.

Nevertheless, after a certain amount of treated images, I get  the `cache resources exhausted` error and the process stops abruptly.

Any idea on how I can overcome the limitations and go through the process?

Thx,
S.

**Comment by asitemade4u** [comment]
Here is the exact error message, as reported by R through Drone CI:
`Error: R: cache resources exhausted `/mnt/stp/53a3fe8a467ddd70108a7e660362a29d/brooklyn-heights-0028.jpg' @ error/cache.c/OpenPixelCache/4095`

**Comment by asitemade4u** [comment]
I have tried to reduce the number of threads as per #346 and it did not work either.

**Comment by jeroen** [comment]
Yes imagemagick uses a lot of memory. You can try running `gc()` somewhere, or explicitly call `image_destroy()` to free memory of images that you no longer need.

**Comment by asitemade4u** [comment]
THANKS @jeroen , `gc()` seems to work fine -- I could treat the batch that was not working previously.
I will try with an even larger number of images and let you know.

**Comment by gavril0** [comment]
I get the same error

```
Error: RGui.exe: UnableToExtendCache `D:\tmp\image_000556.jpg': No space left on device @ error/cache.c/OpenPixelCache/4141
```

when processing frames of a video.   The pipeline and cause of the error is the following:

1. I can extract frames from a video with `av_video_images`, which creates 850 files which have JPEG   3840 x 1920 (72x72) format (246 KB each).  
2. The error occurs when call `image_read` on this vector of images. 
 
My understanding is that magick reads the jpeg files and store them internally in some uncompressed format. It also caches them on the hard disk when memory is exhausted.  This fills up the the 25 GB available on the hard disk where I installed magick fill up at frame 556.

I can address the problem by processing the video in small chunk but, @jeroen, I wonder
if there is a way to specific the location of magick cache from R as suggested here?  

https://www.imagemagick.org/script/architecture.php#tera-pixel

Thanks.
